### PR TITLE
Proxy fixes for github_graphql and gitextractor

### LINKED ISSUE #5502: [Bug][github] Graphql and git connections does not use proxy configuration
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

Devlake github connections for git and graphql data doesn't reach github when run behind a corporate http proxy. Instead the connection times out since direct connections are blocked by firewall rules:

```
time="2023-06-13 16:20:24" level=error msg=" [pipeline service] [pipeline #6] [task #48] run task failed with panic
	caused by: run task failed with panic (github.com/apache/incubator-devlake/helpers/pluginhelper/api.CreateAsyncGraphqlClient:71)
	Wraps: (2) Post "https://api.github.com/graphql": dial tcp 140.82.121.6:443: i/o timeout
	Wraps: (3) Post "https://api.github.com/graphql"
	Wraps: (4) dial tcp 140.82.121.6:443
	Wraps: (5) i/o timeout
	Error types: (1) *hintdetail.withDetail (2) *hintdetail.withDetail (3) *url.Error (4) *net.OpError (5) *net.timeoutError"
```
The plugin fails when it tries to get it's remaining graphql quota:
https://github.com/apache/incubator-devlake/blob/main/backend/plugins/github_graphql/impl/impl.go#L172
Which uses the oauth2 client created here:
https://github.com/apache/incubator-devlake/blob/main/backend/plugins/github_graphql/impl/impl.go#L163

I tried to fix this by changing the contexts `http.Client` as explained here:
https://go.googlesource.com/oauth2/+/refs/heads/master/oauth2.go#332
However my experience with go is limited so I could never get it to work.

As a workaround we set `HTTP_PROXY` environment variables to configure both git and graphql for now.

### What do you expect to happen

Both github graphql and git should be able to reach github throught the proxy.

### How to reproduce

Block direct traffic to github and force the application to go through an http_proxy and the graphql and git collector fail.

### Anything else

_No response_

### Version

v0.17.0-beta5

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by albinvass**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [ ] I have added relevant tests.
- [ ] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
Fix for github_graphql and gitextractor when running devlake behind a corporate proxy. Currently both connections ignores the proxy configuration option and times out when trying to connect to github.

#### gitextractor
The proxy option `git.ProxyTypeAuto` in libgit2 seems to only use proxy when http proxy environment variables are set, instead `git.ProxyTypeSpecified`should be used to correctly specify proxy url.

#### github_graphql
To set proxy config on oauth2 one has to pass a custom `http.Client` object to the library through the context when creating the oauth2 client:
https://cs.opensource.google/go/x/oauth2/+/refs/tags/v0.9.0:oauth2.go;l=334

### Does this close any open issues?
Closes #5502

### CODE REVIEW

File: backend/plugins/github_graphql/impl/impl.go [hunk 1]
```diff
@@ -166,7 +167,30 @@ func (p GithubGraphql) PrepareTaskData(taskCtx plugin.TaskContext, options map[s
 	src := oauth2.StaticTokenSource(
 		&oauth2.Token{AccessToken: tokens[0]},
 	)
-	httpClient := oauth2.NewClient(taskCtx.GetContext(), src)
+	oauthContext := taskCtx.GetContext()
```
Comment (keon94) [inline backend/plugins/github_graphql/impl/impl.go:170]: I suggest refactoring all of this logic into a NewOauth2Client method inside `api_client.go`, and reusing it across these plugins.

**Comment by likyh** [comment]
I think it's OK. Have you test them?

File: backend/plugins/github_graphql/impl/impl.go (hunk 1 above)
Comment (klesh) [reply backend/plugins/github_graphql/impl/impl.go:170]: @albinvass Nice work, LGTM
@likyh , please take a look if the fix is legit
@keon94 Agreed, but it is out of the scope of this PR. Would you like to create a issue and we can do it later?

**Comment by albinvass** [comment]
I've tested it locally with some iptables configuration, but I'll deploy it in our test environment as a sanity check.

**Comment by albinvass** [comment]
Works in our environment :)

**Comment by klesh** [comment]
hi, @albinvass , The PR looks good, would you mind fixing the linting?

**Review by klesh — APPROVED** [review]
LGTM

**Pull request merged**